dali/test/python/test_fw_iterators.py

```diff
@@ -149,6 +150,43 @@ def test_mxnet_iterator_last_batch_no_pad_last_batch():
     assert len(img_ids_list_set) == data_size
     assert len(set(mirrored_data)) != 1
 
+def test_mxnet_iterator_empty_array():
+    from nvidia.dali.plugin.mxnet import DALIGenericIterator as MXNetIterator
+
+    batch_size = 4
+    size = 5
+
+    def get_data():
+        images = [np.empty((1, 224, 224, 3), dtype = np.float)] * batch_size
+        bboxes = [np.empty((1, 0, 4), dtype = np.uint8)] * batch_size
+        labels = [np.array((1, 0, 1), dtype = np.longlong)] * batch_size
```

Comment:
Why not the fixed size type? (TBH, I even didn't know that there is such thing as np.longlong :C)
```suggestion
        labels = [np.array((1, 0, 1), dtype = np.int64)] * batch_size
```

Reply:
Especially that the docs says it's platform dependent (probably it won't ever be different than 64 bits), but still you're comparing it later to fixed size one.

Reply:
Didn't spot the previous discussion.

Reply:
When you change np.array to np.empty is crashes:
```
File "/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/ndarray.py", line 140, in _new_alloc_handle
    ctypes.c_int(int(_DTYPE_NP_TO_MX[np.dtype(dtype).type])),
KeyError: <class 'numpy.longlong'>
```
It looks that np.array((1, 0, 1), dtype = np.longlong) doesn't propagate through DALI as an empty one after all.

Reply:
It looks like mxnet doesn't have a mapping for
np.longlong in the dict _DTYPE_NP_TO_MX
Only a subset of fixed-size np types that are supported in mxnet:
https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/ndarray/ndarray.py#L63

Reply:
So the error is not us but mxnet not supporting ths datatype.

Reply:
It looks like. We need to handle this somehow.

Reply:
Fixed... somehow